src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py

```diff
@@ -225,6 +225,29 @@ def set_aligned_segment_starting_times(self, aligned_segment_starting_times: Lis
             ]
             self.set_aligned_segment_timestamps(aligned_segment_timestamps=aligned_segment_timestamps)
 
+    def set_probe(self, probe, *, group_mode: Union[Literal["by_shank"], Literal["by_probe"]]):
+        """
+        Set the probe information via a ProbeInterface object.
+
+        Parameters
+        ----------
+        probe : probeinterface.Probe
+            The probe object.
+        group_mode : {'by_shank', 'by_probe'}
+            How to group the channels. If 'by_shank', channels are grouped by the shank_id column.
+            If 'by_probe', channels are grouped by the probe_id column.
+            This is a required parameter to avoid the pitfall of using the wrong mode.
+        """
+        # Set the probe to the recording extractor
+        self.recording_extractor.set_probe(
+            probe,
+            in_place=True,
+            group_mode=group_mode,
+        )
+        # Spike interface sets the "group" property
+        # Buy neuroconv expects "group_name" to be set
+        self.recording_extractor.set_property("group_name", self.recording_extractor.get_property("group").astype(str))
```

Comment:
If you really want to keep the `group_name` stuff here, then I'd recommend at least

```python
if "group_name" in recording.get_property_keys():
    self.recording_extractor.set_property("group_name", self.recording_extractor.get_property("group").astype(str))
```

to avoid adding it unnecessarily when it hasn't been set during interface initialization

But I wonder, have you tried the deletion method yet? I think it would produce same group structure result but one less column of duplicated info on the table

Reply:
The problem with the deletion method is that we need to also convert int to str

Reply:
Oh no, channel groups in SI are still allowed to be `int`? So `group_name` is still needed after all it seems... sounds fine then